experimental/uefi/channel/src/lib.rs

```diff
@@ -32,25 +33,76 @@ pub mod schema {
 }
 
 /// Length of the entire frame, including the header.
-pub type FrameLength = u32;
+type FrameLength = u32;
 const FRAME_LENGTH_SIZE: usize = 4;
 static_assertions::assert_eq_size!([u8; FRAME_LENGTH_SIZE], FrameLength);
 
-/// In request frames this field contains the id of the method_or_status that is invoked.
-/// In response frames this field contains the status code of the method_or_status invocation.
-pub type Method = u32;
+/// In request frames this field contains the id of the method that is invoked.
+/// In response frames this field contains the status code of the method invocation.
+type MethodOrStatus = u32;
 const METHOD_SIZE: usize = 4;
-static_assertions::assert_eq_size!([u8; METHOD_SIZE], Method);
+static_assertions::assert_eq_size!([u8; METHOD_SIZE], MethodOrStatus);
 
-pub const FRAME_HEADER_SIZE: usize = 8;
+// TODO(#2848): Use this space to send a stream identifier that identifies
+// frames as belonging to the same message. Currently the frames of every message
+// must be sent as an uninterrupted seqeuence. A stream identifier would allow us
+// to interleave messages.
+/// Reserved space to maintain 8 byte alignement.
+const SPACER_SIZE: usize = 4;
+
+#[repr(u32)]
+#[derive(Copy, Clone)]
+enum Flag {
+    /// Atomic frame, a message sent as a single frame
+    AtomicFrame = 0,
+    /// Starts a stream of frames that constitute a message
+    StreamStart = 1,
+    /// Continues a stream of frames that constitute a message
+    StreamContinuance = 2,
+    /// Ends a stream of frames that constitute a message, indicating that the recipient may
+    /// process the message.
+    StreamEnd = 3,
+}
+const FLAG_SIZE: usize = 4;
+static_assertions::assert_eq_size!([u8; FLAG_SIZE], Flag);
+impl TryFrom<[u8; FLAG_SIZE]> for Flag {
+    type Error = anyhow::Error;
+
+    fn try_from(value: [u8; FLAG_SIZE]) -> Result<Self, Self::Error> {
+        let flag = match u32::from_le_bytes(value) {
+            0 => Flag::AtomicFrame,
+            1 => Flag::StreamStart,
+            2 => Flag::StreamContinuance,
+            3 => Flag::StreamEnd,
+            _ => anyhow::bail!("invalid flag"),
```

Comment:
There must be a way of performing this automatically? Otherwise what's the point of `repr(u32)`? I think a [numeric cast](https://doc.rust-lang.org/reference/expressions/operator-expr.html#semantics) should work, or if not, perhaps see https://enodev.fr/posts/rusticity-convert-an-integer-to-an-enum.html .

Reply:
We already have `strum` as a dependency and use it elsewhere in the codebase for converting between enums and its numeric representations. https://github.com/project-oak/oak/blob/7596b65f92f0bd5f86b9e9d2a1804c41ef58129e/experimental/uefi/kernel/src/boot.rs#L17-L19

Reply:
thanks, updated to use strum